docs/design/supporting-multi-arch.md

```diff
@@ -0,0 +1,137 @@
+# Supporting OCI bytecode Images across Multiple Architectures
+
+Today bpfman has [defined a spec](../developer-guide/shipping-bytecode.md) for shipping eBPF programs via OCI images, however
+it assumes the image builder explicitly knows the operating system
+architecture where the program will be deployed.  Ultimately the spec needs to
+support running on **any** valid linux architecture, this doc will describe
+the changes needed for the specification to support such a task.
+
+## Building eBPF programs for multiple architectures
+
+eBPF programs only have two different completion targets for big or little endian systems.
+This can be clearly seen with the clang compiler:
+
+```console
+clang -print-targets
+
+  Registered Targets:
+    ...
+    bpf         - BPF (host endian)
+    bpfeb       - BPF (big endian)
+    bpfel       - BPF (little endian)
+    ...
+```
+
+Therefore, with those two targets a single eBPF program, which is written
+following [current CO-RE best practices](https://developers.redhat.com/articles/2023/10/19/ebpf-application-development-beyond-basics), can essentially be compiled
+to run successfully on any applicable linux target. However, occasionally
```

Comment:
Citation needed 😆 
We know tracing programs - i.e those that use pt_regs to access CPU registers - will never be cross-platform, regardless of whether you use CO:RE or not.
There are various other edge cases too, like a certain program type, bpf helper or feature not being available in a given architecture too.